checks/system/slurm/slurm.py

```diff
@@ -197,6 +196,11 @@ class MemoryOverconsumptionCheck(SlurmCompiledBaseCheck):
     sourcepath = 'eatmemory.c'
     executable_opts = ['4000M']
 
+    @run_before('compile')
```

Comment:
```suggestion
    @run_after('init')
```
I think it's better we do it before setup, there is no need to create the stage directory for this test. We could also re-write valid systems but skipping may is simpler.